client/src/components/FloorDetailsForm.tsx

```diff
@@ -32,9 +32,9 @@ const FloorDetailsForm: React.FC<FloorDetailsFormProps> = ({
   floorTagState,
   submitVisibilityState,
 }): JSX.Element => {
-  if (showForm === undefined) showForm = true;
+  const [user] = useUserContext();
+  if (showForm === undefined && user?.isAdmin) showForm = true;
```

Comment:
Maybe we can shorten this condition by:

`if (!showForm && user?.isAdmin)`

Another way could be:

`showForm = !showForm && user?.isAdmin` (Assuming they are meant to equal to true)

Reply:
fixed...

Reply:
Hi, I know I'm not in work but just saw this and felt that I needed to comment real quick
`if (!showForm && user?.isAdmin)` will not work as intended since showForm is meant to be a boolean. There are some cases where the `showForm` prop can be false and in this case, we **don't want** the form to be shown. The reason the condition is `=== undefined` is that when the prop (which is an _optional_ prop when not given as the component argument, it then default set it to `true`.

Highly suggest the component to be `showForm === undefined`

Reply:
seems, even if showForm true, When user is not admin, the form will not display.... is what expected...

> Hi, I know I'm not in work but just saw this and felt that I needed to comment real quick `if (!showForm && user?.isAdmin)` will not work as intended since showForm is meant to be a boolean. There are some cases where the `showForm` prop can be false and in this case, we **don't want** the form to be shown. The reason the condition is `=== undefined` is that when the prop (which is an _optional_ prop i.e. when not given as the component argument) is not provided, it then will default set it to `true`, showing the form regardless.
> 
> Highly suggest the component to be reverted back to `showForm === undefined`